internal/service/service.go

```diff
@@ -96,7 +96,12 @@ func (s *Service) Run() error {
 	if err != nil {
 		return fmt.Errorf("failed to listen on tcp port %d", s.Port)
 	}
-	defer ln.Close()
+	defer func(ln net.Listener) {
+		err := ln.Close()
+		if err != nil {
+			s.Logger.WithError(err).Error("failed to close listener")
+		}
+	}(ln)
 	tlsListener := tls.NewListener(ln, config)
```

Comment:
@baoy1 is this change to fix some golang version upgrade related issue? What if not changing it?

Reply:
That fixes gosec error:
```
 G307 (CWE-703): Deferring unsafe method "Close" on type "net.Listener" (Confidence: HIGH, Severity: MEDIUM)
```

Reply:
got it. thanks for explaining it.